TopNews/app/src/main/java/com/fisfam/topnews/MainActivity.java

```diff
@@ -89,6 +96,35 @@ public boolean onOptionsItemSelected(MenuItem item) {
         return super.onOptionsItemSelected(item);
     }
 
+    //Choose country of news source
+    private void chooseCountry() {
+        mUserPref = new UserPreference(getApplicationContext());
+        // setup the alert builder
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setTitle("Choose a country:");
+        // add a radio button list
+        String[] countries = {"Germany", "France", "England", "USA", "China", "Italy"};
+        //the button is always checked by the value stored in mUserPref
+        builder.setSingleChoiceItems(countries, Arrays.asList(countries).indexOf(mUserPref.getCountry()),
+                new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                mUserPref.setCountry(countries[which]);
+            }
+        });
+        // add Done button to close dialog and reload HomeFragment
+        builder.setPositiveButton("Done", new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                dialog.dismiss();
+                loadFragment(new HomeFragment());
```

Comment:
Fine for now but maybe we don't need to create the home fragment again, just refresh the current one

Reply:
Lam the nao de dang tu MainActivity di toi HomeFragment va dung method showRefreshing() and requestData?

Reply:
plz use fking google: https://stackoverflow.com/questions/14247954/communicating-between-a-fragment-and-an-activity-best-practices